File: 2934-minimum-operations-to-maximize-last-elements-in-arrays/2934-minimum-operations-to-maximize-last-elements-in-arrays.cpp
class Solution {
public:
    //Time complexity: O(n)
    //Space complexity: O(1)
    int minOperations(vector<int>& A, vector<int>& B) {
        int ans = 0;
        int n = A.size();
        
        auto count = [&]() {
            int ans = 0;
            
            for(int i = n - 2;i >= 0;i--) {
                if(A[n - 1] < A[i]) {
                    if(B[i] > A[n - 1] || A[i] > B[n - 1]) {
                        return -1;
                    }
                    ans++;
                } else if(B[n - 1] < B[i]) {
                    if(B[i] > A[n - 1] || A[i] > B[n - 1]) {
                        return -1;
                    }
                    ans++;
                }
            }
            
            return ans;
        };
        
        int ans1 = count(), ans2 = -1;
        swap(A[n - 1], B[n - 1]);
        ans2 = count();
        if(ans2 != -1) ans2++;
        
        if(ans1 == -1) return ans2;
        if(ans2 == -1) return ans1;
        
        return min(ans1, ans2);
    }
};

File: 2934-minimum-operations-to-maximize-last-elements-in-arrays/2934-minimum-operations-to-maximize-last-elements-in-arrays.py
class Solution:
    def minOperations(self, nums1: List[int], nums2: List[int]) -> int:
        if nums1[-1] > nums2[-1]:
            nums1, nums2 = nums2, nums1
        max1, max2 = nums1[-1], nums2[-1]
        rs_swap_last_element = 1
        rs = 0
        for i in range(len(nums1) - 1):
            if max(nums1[i], nums2[i]) > max2 or min(nums1[i], nums2[i]) > max1:
                return -1
            elif nums1[i] > max1:
                rs += 1
            elif nums2[i] > max1:
                rs_swap_last_element += 1
        return min(rs, rs_swap_last_element)